File: bindings/flutter/android/src/main/kotlin/io/github/alexdonh/hologramism/flutter/HologramismPlugin.kt
package io.github.alexdonh.hologramism.flutter

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Registers the native platform view that backs the Dart `HologramView` widget. */
class HologramismPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        binding.platformViewRegistry.registerViewFactory(
            "hologramism/HologramView",
            HologramViewFactory(binding.binaryMessenger),
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: bindings/flutter/ios/Classes/HologramismPlugin.swift
import Flutter
import UIKit

/// Registers the native platform view that backs the Dart `HologramView` widget.
public class HologramismPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let factory = HologramViewFactory(messenger: registrar.messenger())
    registrar.register(factory, withId: "hologramism/HologramView")
  }
}
